go/signedexchange/cbor/encoder_test.go

```diff
@@ -46,10 +47,139 @@ func TestIntegers(t *testing.T) {
+		encoding string
+	}{
+		{[]byte{}, "60"},
+		{[]byte{0xab}, "41ab"},
+		{
+			[]byte{0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19, 20, 21, 22, 23, 24},
+			"5818 00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f 10 11 12 13 14 15 16 17",
+		},
+	}
+	// This doesn't test every size of integer that might encode the length
+	// of the byte string.
+
+	for _, test := range textTests {
+		var b bytes.Buffer
+		e := &Encoder{&b}
+
+		if err := e.EncodeByteString(test.bs); err != nil {
+			t.Errorf("Encode. err: %v", err)
+		}
+		exp := fromHex(test.encoding)
+
+		if !bytes.Equal(exp, b.Bytes()) {
+			t.Errorf("%v expected to encode to %v, actual %v", test.bs, exp, b.Bytes())
+		}
+	}
+}
+
+func TestEncodeTextString(t *testing.T) {
+	var textTests = []struct {
+		s        string
+		encoding string
+	}{
+		{"", "60"},
+		{"a", "6161"},
+		{"IETF", "6449455446"},
+		{`"\`, "62225c"},
+		{"\u00fc", "62c3bc"},
+		{"\u6c34", "63e6b0b4"},
+		{"\U00010151", "64f0908591"},
+	}
+	for _, test := range textTests {
+		var b bytes.Buffer
+		e := &Encoder{&b}
+
+		if err := e.EncodeTextString(test.s); err != nil {
+			t.Errorf("Encode. err: %v", err)
+		}
+		exp := fromHex(test.encoding)
+
+		if !bytes.Equal(exp, b.Bytes()) {
+			t.Errorf("\"%s\" expected to encode to %v, actual %v", test.s, exp, b.Bytes())
+		}
+	}
+}
+
+func TestMapEncoderSorter(t *testing.T) {
+	/*
+	  The keys in every map MUST be sorted in the bytewise lexicographic order of their canonical encodings. For example, the following keys are correctly sorted:
+	  1. 10, encoded as 0A.
+	  2. 100, encoded as 18 64.
+	  3. -1, encoded as 20.
+	  4. “z”, encoded as 61 7A.
+	  5. “aa”, encoded as 62 61 61.
+	  6. [100], encoded as 81 18 64.
+	  7. [-1], encoded as 81 20.
+	  8. false, encoded as F4.
+	*/
+
+	es := []*MapEntryEncoder{
+		GenerateMapEntry(func(keyE Encoder, valueE Encoder) {
+			keyE.EncodeInt(10)
+			valueE.EncodeTextString("int 10")
+		}),
+		GenerateMapEntry(func(keyE Encoder, valueE Encoder) {
+			keyE.EncodeInt(100)
+			valueE.EncodeTextString("int 100")
+		}),
+		GenerateMapEntry(func(keyE Encoder, valueE Encoder) {
+			keyE.EncodeInt(-1)
+			valueE.EncodeTextString("int -1")
+		}),
+		GenerateMapEntry(func(keyE Encoder, valueE Encoder) {
+			keyE.EncodeTextString("z")
+			valueE.EncodeTextString("string \"z\"")
+		}),
+		GenerateMapEntry(func(keyE Encoder, valueE Encoder) {
+			keyE.EncodeTextString("aa")
+			valueE.EncodeTextString("string \"aa\"")
+		}),
+		GenerateMapEntry(func(keyE Encoder, valueE Encoder) {
+			keyE.EncodeArrayHeader(1)
+			keyE.EncodeInt(100)
+			valueE.EncodeTextString("array [100]")
+		}),
+		GenerateMapEntry(func(keyE Encoder, valueE Encoder) {
+			keyE.EncodeArrayHeader(1)
+			keyE.EncodeInt(-1)
+			valueE.EncodeTextString("array [-1]")
+		}),
+		GenerateMapEntry(func(keyE Encoder, valueE Encoder) {
+			keyE.EncodeBool(false)
+			valueE.EncodeTextString("false")
+		}),
+	}
+
+	keyExp := [][]byte{
+		fromHex("0A"),
```

Comment:
I'd write `{0x0a}`. Is this really needed to convert bytes from string literals?